lambda/aws-rag-appsync-stepfn-opensearch/embeddings_job/src/lambda.py

```diff
@@ -151,54 +156,75 @@ def handler(event,  context: LambdaContext) -> dict:
+        files.append({'name':transformed_file['name'],
+                       'status':transformed_file['s3_transformer_result']['Payload']['status'],
+                       'imageurl':''})
     updateIngestionJobStatus({'jobid': job_id, 'files': files})
 
-    print(f'Loading txt raw files from {bucket_name}')
+    
+    
+    logger.info(f'Loading txt raw files from {bucket_name}')
 
     docs = []
+    
+    # Images are stored in s3 with presigned url, embeddings is not required.
 
     for transformed_file in event:
         if transformed_file['s3_transformer_result']['Payload']['status'] == 'File transformed':
             filename = transformed_file['s3_transformer_result']['Payload']['name']
-            loader = S3TxtFileLoaderInMemory(bucket_name, filename)
-            sub_docs = loader.load()
-            for doc in sub_docs:
-                doc.metadata['source'] = filename
-            docs.extend(sub_docs)
+            original_filename = transformed_file['name']
+            name, extension = os.path.splitext(original_filename)
+            print(f" the original_filename {name} and extension {extension}")
+            if(extension == '.pdf'):
+                loader = S3TxtFileLoaderInMemory(bucket_name, filename)
+                sub_docs = loader.load()
+                for doc in sub_docs:
+                    doc.metadata['source'] = original_filename
+                docs.extend(sub_docs)
+                process_text_embeddings(docs,modelid,http_auth,files,job_id)
+            if(extension == '.jpg' or extension == '.jpeg' or extension == '.png' or extension == '.svg'):
+                img_load = image_loader(bucket_name, filename,f"{name}.txt",modelid)
+                docs = img_load.load()
+                url=img_load.get_presigned_url()
+                for doc in docs:
+                    doc.metadata['image_path'] = url
+                process_image_embeddings(docs,modelid,http_auth,files,job_id,url)
 
     if not docs:
-        return {
-            'status':'nothing to ingest'
-        }
+            return {
+                'status':'nothing to ingest'
+            }
+
+   
 
+def process_text_embeddings(docs,modelid,http_auth,files,job_id):
+    logger.info("process image embeddings with chunks")
     text_splitter = RecursiveCharacterTextSplitter(
                 # Set a really small chunk size, just to show.
                 chunk_size=CHUNCK_SIZE_DOC_SPLIT,
                 chunk_overlap=OVERLAP_FOR_DOC_SPLIT,
                 length_function=len,
             )
 
-    print('Documents loaded locally')
-
     # add a custom metadata field, such as timestamp
     # we can augment data here probably (PII present ? ...)
     for doc in docs:
         doc.metadata['timestamp'] = time.time()
-        doc.metadata['embeddings_model'] = 'amazon.titan-embed-text-v1'
+        doc.metadata['embeddings_model'] = modelid
```

Comment:
have you tested to use other embeddings models from Bedrock ?

Reply:
yes , tested with 'amazon.titan-embed-text-v1' and 'amazon.titan-embed-image-v1'